ravenpy/utilities/gis.py

```diff
@@ -3,35 +3,146 @@
 -----------------------------------
 
 * Point coordinates are passed as shapely.geometry.Point instances.
-* BBox coordinates are passed as (lon1, lat1, lon2, lat2)
+* BBox coordinates are passed as (lon1, lat1, lon2, lat2).
 * Shapes (polygons) are passed as ?
-* All functions that require a CRS have a CRS argument with a default set to WSG84 (?)
-*
+* All functions that require a CRS have a CRS argument with a default set to WSG84.
+* GEO_URL points to the GeoSever hosting all files.
 
 """
-
 import collections
 from pathlib import Path
 from typing import Iterable, List, Sequence, Tuple, Union
 
 import fiona
 import pandas as pd
+from lxml import etree
+from owslib.fes import PropertyIsLike
+from owslib.wfs import WebFeatureService
+from requests import Request
 from shapely.geometry import Point, shape
 
 from ravenpy.utils import crs_sniffer, single_file_check
 
-GEO_URL = "http://boreas.ouranos.ca/geoserver/wfs"
+GEO_URL = "http://boreas.ouranos.ca/geoserver/"
 
 # We store the contour of different hydrobasins domains
 hybas_dir = Path(__file__).parent.parent / "data" / "hydrobasins_domains"
 hybas_pat = "hybas_lake_{}_lev01_v1c.zip"
 
 # This could be inferred from existing files in hybas_dir
 hybas_regions = ["na", "ar"]
-
 hybas_domains = {dom: hybas_dir / hybas_pat.format(dom) for dom in hybas_regions}
 
 
+def _get_location_wfs(
```

Comment:
The name suggests this is a generic function, whereas it looks tied to a very specific dataset.